python/ray/tune/execution/tune_controller.py

```diff
@@ -1602,6 +1602,19 @@ def _schedule_graceful_trial_stop(self, trial: Trial):
             self._schedule_trial_stop(trial)
 
     def _schedule_trial_pause(self, trial: Trial, should_checkpoint: bool = True):
+        if _use_storage_context():
+            if should_checkpoint:
+
+                def on_result(*args, **kwargs):
+                    self._on_saving_result(*args, **kwargs)
+                    self._schedule_trial_stop(trial)
+                    self._set_trial_status(trial, Trial.PAUSED)
```

Comment:
Instead of providing a custom callback here, can we call `self._schedule_trial_task` here? It seems we only have to copy the `if trial not in self._trial_to_actor:` line from `_schedule_trial_save`

Reply:
Done, lmk if this is what you were thinking.